### PR TITLE
Be nc news ticket04

### PR DESCRIPTION
**Pull request by TypeError92**

### CODE REVIEW

File: __tests__/endpoints.test.js [hunk 1]
```diff
@@ -25,6 +26,57 @@ describe('GET /api', () => {
...
+            .then(({body}) => {
+                expect(Object.keys(body)).toEqual([
+                    'article_id',
+                    'title',
+                    'topic',
+                    'author',
+                    'body',
+                    'created_at',
+                    'votes',
+                    'article_img_url'
```
Comment (johnhutcheon) [inline __tests__/endpoints.test.js:44]: Just a note- later on this endpoint will be adding a further key. I'm pretty sure that this test will still pass the way you've done it, but if not, you may have to come back and change this to a more flexible matcher later 😄

File: __tests__/endpoints.test.js [hunk 2]
```diff
@@ -25,6 +26,57 @@ describe('GET /api', () => {
...
+                ])
+
+                expect(body.article_id).toBe(1)
```
Comment (johnhutcheon) [inline __tests__/endpoints.test.js:47]: Good work checking the specific article number 👍

File: controllers/error-handlers.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+const psqlErrorCodes = {
+    '22P02': {status: 400, msg: 'invalid_text_representation'}
+}
```
Comment (johnhutcheon) [inline controllers/error-handlers.js:3]: I'd maybe change this to something slightly more user friendly like 'Invalid Input', but I'll leave it up to you to decide 🙂

File: models/articles.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    return db.query(
+        `
+        SELECT * FROM articles
+        WHERE article_id = $1;
+        `,
+        [article_id]
+    )
+    .then(({rows}) => {
+        return rows.length ? rows[0] : Promise.reject({status: 404, msg: 'ID not found'})
+    })
```
Comment (johnhutcheon) [inline models/articles.js:13]: Nicely handled 👍

**Review by johnhutcheon — APPROVED** [review]
Hey Simon, nice work! I've left a couple of suggestions for you to look at, but you're all good to merge and move on. Well done! 🙌

File: controllers/error-handlers.js (hunk 3 above)
Comment (TypeError92) [reply controllers/error-handlers.js:3]: This is a actually the description psql provides for that error code, I agree that it's not intuitive. Maybe it's worth waiting a bit until I've collected all relevant psql errors and then group them into something more natural?

File: __tests__/endpoints.test.js (hunk 1 above)
Comment (TypeError92) [reply __tests__/endpoints.test.js:44]: Thanks for the tip, I'll keep that in mind ;)

**Pull request merged**